build.psm1

```diff
@@ -2514,6 +2514,7 @@ function Restore-PSModule
                         Name=$_
                         Path = $Destination
                         Repository =$RepositoryName
+                        AllowPreRelease = $true
```

Comment:
This is causing failures

Reply:
Yes, I know, see my request for feedback [here](https://github.com/PowerShell/PowerShell/pull/5759#issuecomment-354506893).

Reply:
Also, We should only do this if needed.